### ISSUE TITLE
Documentation is out of date

### ISSUE THREAD

**Issue by illright** [encode/django-rest-framework#6132]
The documentation on the website seems to be out of date.

For instance, [here](https://www.django-rest-framework.org/tutorial/4-authentication-and-permissions/#adding-login-to-the-browsable-api) you can see this code snippet:
```python
urlpatterns += [
    url(r'^api-auth/', include('rest_framework.urls')),
]
```
which is wrong, it should be `path()`, not `url()`. However, it seems like in the master branch this issue is fixed. So I guess we need to push the docs changes to the website.

**Comment by norac89** [comment]
Also seems this issue is part of milestone 3.8.3, which looks completed and merged but has not received a release tag.

**Comment by carltongibson** [comment]
`url()` is still perfectly valid. (It's not going anywhere, even if `path` is the future.) 

The v3.8.3 milestone will be rolled into v3.9, which will be the next release in ≈one month.

**Comment by morenoh149** [comment]
for those seeking an example for this the best I could come up with was
```py
# urls.py
from django.urls import include, path                                               
from rest_framework import routers                                                  
                                                                                    
from quickstart import views                                                             
                                                                                    
router = routers.DefaultRouter()                                           
router.register(r'users', views.UserViewSet)                                     
router.register(r'groups', views.GroupViewSet)                                      
                                                                                    
urlpatterns = [                                                                                      
    path('', include(router.urls)),                                                 
    path('api-auth/', include('rest_framework.urls', namespace='rest_framework'))    
]         
```

**Comment by HarmvZ** [comment]
Why is this issue closed? The documentation on [the official site](http://www.django-rest-framework.org) is still out of date. The docs on github are up to date however.

**Comment by tomchristie** [comment]
@HarmvZ Because `url` is valid. If this was a bug then we might roll out a docs fix earlier. As it currently stands it'll wait a while longer until next upcoming release, which is due shortly.

**Comment by HarmvZ** [comment]
@tomchristie I agree, it is indeed not a bug. But the issue is about the general fact that the docs are out of date. For instance, if you look under requirements on the homepage you would think Python 3.7 is still not supported.

**Comment by tomchristie** [comment]
Typically we’d have the docs up to date. However we’ve erroneously merged some docs only applicable to 3.8 which isn’t released yet, and the simplest thing for us to do is for us to wait a week or so before the next release.

**Comment by carltongibson** [comment]
The published docs are for the last release 3.8.2. They’re correct for that. Changes in master since then won’t be reflected until the next release (as Tom says). I can’t see what’s meant to be wrong. 

E.g. Python 3.7 wasn’t tested against, nor did we claim to support it for 3.8.2. There’s no reason to say it’s supported.

**Comment by HarmvZ** [comment]
Okay, I understand. Thanks for the info!

**Comment by tomchristie** [comment]
> The published docs are for the last release 3.8.2. They’re correct for that. Changes in master since then won’t be reflected until the next release (as Tom says). I can’t see what’s meant to be wrong.

Gotcha. My misunderstanding.
